util/dvsim/SimCfg.py

```diff
@@ -507,13 +526,18 @@ def _create_deploy_objects(self):
                 test.build_mode = Modes.find_mode(
                     build_map[test.build_mode].name, self.build_modes)
 
-        if self.run_only:
-            self.builds = []
-            build_map = None
-
         self.runs = ([]
                      if self.build_only else self._expand_run_list(build_map))
 
+        # Discard the build_job dependency that was added earlier if --run-only
+        # switch is passed. This is a little cumbersome than previous approach,
+        # but needed so that we can accurately set the index and total for each
+        # run job.
+        if self.run_only:
+            self.builds = []
+            for run in self.runs:
+                run.dependencies.pop()
```

Comment:
This removes exactly one of the dependencies. Is that intentional? Couldn't we just have `run.dependencies = []`?

Reply:
Yeah, I guess, but it seems more obvious to clear dependencies than to rely on the fact that we know the list is a singleton and then remove its one and only element.